pkg/monitor/shutdown/consumer.go

```diff
@@ -0,0 +1,160 @@
+package shutdown
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	Reason                       = "GracefulShutdownWindow"
+	KubeAPIServerShutdownLocator = "shutdown/apiserver"
```

Comment:
any suggestion on the locator name?

Reply:
Locator name seems to be fine for me. Maybe rename the variable KubeAPIServerShutdownLocator to APIServerShutdownLocator since this is not specific to kube-apiserver. Nit though.